src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -405,36 +405,45 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
         return new Tuple<bool, string>(success, paymentIntentClientSecret);
     }
 
-    private async Task ValidateSecretsManagerSeatsAndServiceAccount(OrganizationUpgrade upgrade, Organization organization, short newPlanSeats, Models.StaticStore.Plan newSecretsManagerPlan)
+    private async Task ValidateSecretsManagerSeatsAndServiceAccountAsync(OrganizationUpgrade upgrade, Organization organization,
+        Models.StaticStore.Plan newSecretsManagerPlan)
     {
         var newPlanSmSeats = (short)(newSecretsManagerPlan.BaseSeats +
                                      (newSecretsManagerPlan.HasAdditionalSeatsOption
                                          ? upgrade.AdditionalSmSeats
                                          : 0));
+        var occupiedSmSeats =
+            await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
+
         if (!organization.SmSeats.HasValue || organization.SmSeats.Value > newPlanSmSeats)
         {
-            var occupiedSmSeats =
-                await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
-            if (occupiedSmSeats > newPlanSeats)
+            if (occupiedSmSeats > newPlanSmSeats)
             {
                 throw new BadRequestException(
                     $"Your organization currently has {occupiedSmSeats} Secrets Manager seats filled. " +
-                    $"Your new plan only has ({newPlanSeats}) seats. Remove some users.");
+                    $"Your new plan only has ({newPlanSmSeats}) seats. Remove some users.");
             }
         }
 
+        var occupiedPmSeats = await _organizationUserRepository.GetOccupiedSeatCountByOrganizationIdAsync(organization.Id);
+
+        if (occupiedSmSeats > occupiedPmSeats)
```

Comment:
We're not concerned with occupied seats here (which is the number of active users). We're concerned about what they've added to the plan (what they're buying). It doesn't make sense to _buy_ more SM seats than PM seats.

So this can just be `if (upgrade.AdditionalSmSeats > upgrade.AdditionalSeats)`. You can remove these repository calls you added.

You also need to do this check whenever SM seat count changes:
* create flow
* upgrade flow
* adjust subscription flow

Reply:
I will apply it on adjust subscription on the AC-1410